pkg/cli/cluster_free_disk_space.go

```diff
@@ -0,0 +1,232 @@
+package cli
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"io"
+	"log"
+	"os/signal"
+	"syscall"
+
+	storagev1 "k8s.io/api/storage/v1"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client/config"
+
+	"code.cloudfoundry.org/bytefmt"
+	rookcli "github.com/rook/rook/pkg/client/clientset/versioned"
+	"github.com/spf13/cobra"
+
+	clusterspace "github.com/replicatedhq/kurl/pkg/cluster/space"
+)
+
+const (
+	// defaultOpenEBSPodImage is the image used during disk free check for openebs storage. this is the same image used by the
+	// pvmigrate project, it may be any image containing 'df' and 'cat' commands.
+	defaultOpenEBSPodImage          = "eeacms/rsync:2.3"
+	isDefaultStorageClassAnnotation = "storageclass.kubernetes.io/is-default-class"
+	openEBSLocalProvisioner         = "openebs.io/local"
+	rookRBDProvisioner              = "rook-ceph.rbd.csi.ceph.com"
+	rookCephFSProvisioner           = "rook-ceph.cephfs.csi.ceph.com"
+)
+
+// getStorageClassByName returns a storage class by its name. if storageClassName is empty then this function returns the default storage
+// class for the cluster.
+func getStorageClassByName(ctx context.Context, kubeCli kubernetes.Interface, storageClassName string) (*storagev1.StorageClass, error) {
+	classes, err := kubeCli.StorageV1().StorageClasses().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		return nil, fmt.Errorf("failed to list cluster storage classes: %w", err)
+	}
+
+	for _, class := range classes.Items {
+		val, ok := class.Annotations[isDefaultStorageClassAnnotation]
+		if ok && val == "true" && storageClassName == "" {
+			return &class, nil
+		}
+
+		if class.Name == storageClassName {
+			return &class, nil
+		}
+	}
+
+	if storageClassName == "" {
+		return nil, fmt.Errorf("failed to find the default storage class")
+	}
+	return nil, fmt.Errorf("failed to find storage class %q", storageClassName)
+}
+
+// hasEnoughSpace compares if free space is bigger than the requested space. returns a user friendly string representing the output
+// and a bool indicating if there is or not enough room. this function is an auxiliar function so we don't need to keep concatenating
+// the output strings in the evaluateOpenEBSFreeSpace function.
+func hasEnoughSpace(node string, free, requested int64) (string, bool) {
+	requestedString := bytefmt.ByteSize(uint64(requested))
+	freeString := bytefmt.ByteSize(uint64(free))
+	if free >= requested {
+		message := fmt.Sprintf("Node %s has %s available", node, freeString)
+		if requested > 0 {
+			message = fmt.Sprintf("%s (requested %s)", message, requestedString)
+		}
+		return message, true
+	}
+	return fmt.Sprintf("Not enough space on node %s (requested %s, available %s)", node, requestedString, freeString), false
+}
```

Comment:
```suggestion
```

Why we would add that in the check? 
What is the motivation?

Reply:
See my previous answers